serviceAccount/ServiceAccount.psm1

```diff
@@ -0,0 +1,1211 @@
+        Return $false 
+    } 
+ 
+    # Generate UpdateServiceSettings.sql, but not for secondary WID. Secondary SQL never gets to this function 
+     
+    If (!(($Role -eq "SecondaryComputer") -and ($DBMode -eq "WID"))) 
+    { 
+        "USE AdfsConfiguration" | Out-File "$env:Temp\ADFSSQLScripts\UpdateServiceSettings.sql" 
+        "SELECT ServiceSettingsData from IdentityServerPolicy.ServiceSettings" | Out-File "$env:Temp\ADFSSQLScripts\UpdateServiceSettings.sql" -append 
+        "UPDATE IdentityServerPolicy.ServiceSettings" | Out-File "$env:Temp\ADFSSQLScripts\UpdateServiceSettings.sql" -append 
+        "SET ServiceSettingsData=REPLACE((SELECT ServiceSettingsData from IdentityServerPolicy.ServiceSettings),'$OldSID','$NewSID')" | Out-File "$env:Temp\ADFSSQLScripts\UpdateServiceSettings.sql" -append 
+        "SELECT ServiceSettingsData from IdentityServerPolicy.ServiceSettings" | Out-File "$env:Temp\ADFSSQLScripts\UpdateServiceSettings.sql" -append 
+     
+        If (!$?) 
+        { 
+            Write-Host "`tFailed to generate UpdateServiceSettings.sql. Exiting" -ForegroundColor "red" 
+            ($ElapsedTime.Elapsed.ToString())+" [ERROR]     Failed to generate UpdateServiceSettings.sql" | Out-File $LogPath -Append 
+            Return $false 
+        } 
+    } 
+     
+    # Clean up the CreateDB.sql file 
+    If (Test-Path "$env:Temp\ADFSSQLScripts\CreateDB.sql") 
+    { 
+        Remove-Item "$env:Temp\ADFSSQLScripts\CreateDB.sql" 
+    } 
+     
+    Return $true 
+     
+} 
+     
+     
+# Executes the SQL scripts generated by GenerateSQLScripts 
+Function ExecuteSQLScripts 
+{ 
+    Start sqlcmd.exe -ArgumentList "-S $SQLHost -i $env:Temp\ADFSSQLScripts\SetPermissions.sql -o $env:Temp\ADFSSQLScripts\SetPermissions.log" -Wait -WindowStyle Hidden | Out-File $LogPath -Append 
+      
+    If (!$?) 
+    { 
+        Write-Host "`tFailed to execute SetPermissions.sql. Exiting" -ForegroundColor "red" 
+        ($ElapsedTime.Elapsed.ToString())+" [ERROR]     Failed to execute SetPermissions.sql" | Out-File $LogPath -Append 
+        Return $false 
+    } 
+      
+     # Execute UpdateServiceSettings.sql, but not for secondary WID. Secondary SQL never gets to this function. 
+      
+    If (!(($Role -eq "SecondaryComputer") -and ($DBMode -eq "WID"))) 
+    { 
+        Start sqlcmd.exe -ArgumentList "-S $SQLHost -i $env:Temp\ADFSSQLScripts\UpdateServiceSettings.sql -o $env:Temp\ADFSSQLScripts\UpdateServiceSettings.log" -Wait -WindowStyle Hidden | Out-File $LogPath -Append 
+      
+        If (!$?) 
+        { 
+            Write-Host "`tFailed to execute UpdateServiceSettings.sql. Exiting...." -ForegroundColor "red" 
+            ($ElapsedTime.Elapsed.ToString())+" [ERROR]     Failed to execute UpdateServiceSettings.sql" | Out-File $LogPath -Append 
+            Return $false 
+        } 
+    } 
+    Return $true 
+} 
+
+
+
+#Define functions to export
+
+
```

Comment:
Could you add a few pester tests (similar to what we added for the [logging script](https://github.com/Microsoft/adfsLogTools#pester-tests)) to test this function. Mostly simple things like: 

1. Check service properties for SID rule, call this, check service properties again

2. Call with missing user and validate error

3. Etc.